### ISSUE TITLE
space-before-function-paren does not work for anonymous generator functions

### ISSUE THREAD

**Issue by despairblue** [eslint/eslint#5435]
**What version of ESLint are you using?**
2.2.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
* babel-eslint

```js
{
  "rules": {
    "space-before-function-paren": [2, {"anonymous": "always", "named": "always"}]
  }
}
```
**What did you do? Please include the actual source code causing the issue.**
Example:
```js
var bla = function *() {}

bla()
```

**What did you expect to happen?**
I expected it to not lint, because the documentation for the [generator-star-spacing rule](generator-star-spacing.html) says:
> In anonymous function expressions, spacing between the * and the opening parenthesis is not checked. This is checked by the space-before-function-paren rule.

**What actually happened? Please include the actual, raw output from ESLint.**
Eslint is not complaining about the missing space between the `*` and the `(`.

**Comment by platinumazure** [comment]
Can you reproduce this with the default parser?

**Comment by despairblue** [comment]
Is that the one used for the online demo editor? If so, then yes.

On Mon, Feb 29, 2016, 19:18 Kevin Partington <notifications@github.com>
wrote:

> Can you reproduce this with the default parser?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eslint/eslint/issues/5435#issuecomment-190318059>.
>
-- 
Q: Why is this email five sentences or less?
A: http://five.sentenc.es

**Comment by platinumazure** [comment]
The online demo editor uses an older version of the default parser, so I guess I'm not sure if it might have been fixed in the newest espree. Hopefully you or a team member can reproduce with latest espree and hopefully that will give us more information.

Do let us know if you get a chance to try it out with latest ESLint with the default parser. Thanks!

**Comment by alberto** [comment]
Confirmed the rule is explicitly not erroring for anonymous generators:
```
        if (node.generator && !isNamed) {
            return;
        }
```

@eslint/eslint-team which rule should take care of this?

@btmills you wrote the original code and message.

**Comment by mysticatea** [comment]
And I wrote the sentence of the document, but I cannot remember my thought of that time...

**Comment by btmills** [comment]
The only discussion I can find around this is in #1929 on March 3rd of 2015, which indicated that `generator-star` should cover this case. I added an exception to `space-before-function-parentheses` on March 5th. On March 13th, I submitted #1987, which added the `generator-star-spacing` rule, explicitly excluding the same case. (@mysticatea, you're off the hook - that line in the docs dates from my original commit.) Why I excluded them from both, I don't know, and I can't recall or find any other discussion around it. Does the decision in #1929 that this case belongs to `generator-star-spacing` still sound valid to everyone?

**Comment by despairblue** [comment]
Thanks guys. :+1:  Sry for not being able to verify with the default parser in a timely manner.

**Comment by nzakas** [comment]
`generator-star-spacing` sounds right to me
